tests/spec/s-object.js

```diff
@@ -260,6 +260,38 @@ describe('Object', function () {
         });
     });
 
+    describe('Object.getPrototypeOf', function () {
+        it('should return the [[Prototype]] of an object', function () {
+            function Foo() {}
+
+            var proto = Object.getPrototypeOf(new Foo);
+
+            expect(proto).toBe(Foo.prototype);
+        });
+
+        it('should shamone to the `Object.prototype` if `object.constructor` is not a function', function () {
+            function Foo() {}
+            Foo.prototype.constructor = 1;
+
+            var proto = Object.getPrototypeOf(new Foo),
+                fnToString = Function.prototype.toString;
+
+            if (fnToString.call(Object.getPrototypeOf).indexOf('[native code]') < 0) {
```

Comment:
While this works, it makes me cringe - why is it that native code will return `Object.prototype` but the sham will return `Foo.prototype`? I'd rather the sham threw, than return the wrong value.